gen_future_weather_file.py

```diff
@@ -0,0 +1,425 @@
+            #    then all days after feb 29th in a leap year will be shifted by
+            #    one, causing issues with averaged etc.
+            data[time.timetuple().tm_yday - 1].append(
+                                              datapoint[lat_index][long_index]
+                                               )
+
+    return data
+
+
+def avg(lst: list):
+    """ This method returns the average value of a passed list."""
+
+    return sum(lst) / len(lst) if len(lst) != 0 else 0
+
+
+def gen_future_weather_file(lat: float,
+                            long: float,
+                            present_range: range,
+                            future_range: range,
+                            present_climate: str,
+                            future_climate: str,
+                            netcdf_variable: str,
+                            epw_file: str
+                            ):
+    """ gen_future_weather_file(float, float, range, range, str, str, str)
+
+        Regenerates the passed epw file into a weather file represeting future
+        data.
+
+        Args:
+            lat(float): The latitude to read data from climate files.
+
+            long(float): The logitude to read data from teh climate files.
+
+            present_range(range): The range of years that makes up "present"
+                          for this particular run.
+
+            future_range(range): The range of years that makes up "future" for
+                         this particular run.
+
+            present_climate(str): The path to the climate file with "present"
+                         data.
+
+            future_climate(str): The path to the climate file with "future"
+                         data.
+
+            epw_file(str): The path to the epw file to regenerate.
+    """
+
+    # Get the present and future climate data.
+    present_data = get_climate_data(present_climate, lat,
+                                    long, netcdf_variable, present_range)
+
+    future_data = get_climate_data(future_climate, lat, long,
+                                   netcdf_variable, future_range)
+
+    # Get the data from epw file and the headers from the epw.
+    epw_data, headers = get_epw_data(epw_file)
+
+    daily_averages = get_daily_averages(epw_data)
+
+    # Morph the data in the file so that it reflects what it should be in the
+    # future. IE) run the processes required in by the paper.
+    for row_index, data_row in enumerate(epw_data):
+        day = data_row[0].timetuple().tm_yday - 1
+
+        # morph_data currently has no implementation, so the data being passed
+        # right now may not accurately reflect what will actually be passed
+        # upon implementation.
+        epw_data[row_index] = morph_data(data_row, day, present_data,
+                                         future_data, daily_averages)
+
+    # Write the data out to the epw file.
+    write_epw_data(epw_data, headers, epw_file[:-4] + "_future.epw")
```

Comment:
It would be more flexible here to accept the output file name as a parameter to the script.